### PR TITLE
prune check on Trainer fit result

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

prune check on Trainer fit result...

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by s-rog — APPROVED** [review]

File: tests/models/test_horovod.py [hunk 1]
```diff
@@ -358,7 +359,7 @@ def _compute_batch():
 #             limit_train_batches=0.2,
 #             distributed_backend='horovod'
 #         )
-#         results = trainer.fit(model)
+#         trainer.fit(model)
```
Comment (carmocca) [inline tests/models/test_horovod.py:362]: Either undo this change or fix the assert right below

Commented due to flaky test?

File: tests/backends/test_tpu_backend.py [hunk 2]
```diff
@@ -45,9 +46,8 @@ def test_resume_training_on_cpu(tmpdir):
         max_epochs=1,
         default_root_dir=tmpdir,
     )
-    result = trainer.fit(model)
-
-    assert result == 1
+    trainer.fit(model)
+    assert trainer.state == TrainerState.FINISHED, "Training failed with %s" % trainer.state
```
Comment (carmocca) [inline tests/backends/test_tpu_backend.py:50]: Feel like updating them all again into f-strings (more mordern)? :)

```suggestion
    assert trainer.state == TrainerState.FINISHED, f"Training failed with {trainer.state}"
```

File: tests/models/test_restore.py [hunk 3]
```diff
@@ -186,10 +187,10 @@ def test_running_test_pretrained_model_distrib_dp(tmpdir):
 
     # fit model
     trainer = Trainer(**trainer_options)
-    result = trainer.fit(model)
+    trainer.fit(model)
 
     # correct result and ok accuracy
-    assert result == 1, 'training failed to complete'
+    assert trainer.state == TrainerState.FINISHED, "Training failed with %s" % trainer.state
```
Comment (tchaton) [inline tests/models/test_restore.py:193]: Would be good to change them all to f-string.

**Review by tchaton — CHANGES_REQUESTED** [review]

File: tests/backends/test_tpu_backend.py (hunk 2 above)
Comment (tchaton) [reply tests/backends/test_tpu_backend.py:50]: Agree !

**Review by justusschock — APPROVED** [review]

File: tests/backends/test_tpu_backend.py (hunk 2 above)
Comment (Borda) [reply tests/backends/test_tpu_backend.py]: the diff is lazy and always eval but as you wish...

File: tests/models/test_horovod.py (hunk 1 above)
Comment (Borda) [reply tests/models/test_horovod.py]: no idea, ask the author who commented it out :P

**Comment by Borda** [comment]
Done ;]

**Review by carmocca — APPROVED** [review]
